feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { AsmFacade } from '@spartacus/asm/root';
+import { ActiveCartFacade, MultiCartFacade } from '@spartacus/cart/base/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  OCC_CART_ID_CURRENT,
+  User,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'cx-asm-bind-cart',
+  templateUrl: './asm-bind-cart.component.html',
+})
+export class AsmBindCartComponent implements OnInit, OnDestroy {
+  @Input() customer: User;
+  @Input() cartId: FormControl = new FormControl('', [
+    Validators.required,
+    Validators.minLength(1),
+  ]);
+  isCustomerEmulationSessionInProgress$: Observable<boolean>;
+
+  protected subscription = new Subscription();
+
+  constructor(
+    protected globalMessageService: GlobalMessageService,
+    protected asmFacadeService: AsmFacade,
+    protected activeCartFacade: ActiveCartFacade,
+    protected multiCartFacade: MultiCartFacade
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.activeCartFacade.getActiveCartId().subscribe((response: string) => {
+        if (response) {
+          this.cartId.setValue(response);
+        }
+      })
+    );
+  }
+
+  /**
+   * Assign the input cart number to the customer
+   */
+  assignCartToCustomer() {
```

Comment:
Suggestion: In some places, the action is call "assign" cart ( like here ans the i18n keys) and in some other places it's called "bind" cart.  To improve readability, perhaps we can consider a consistent naming.

Reply:
updated to bind*